instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/event_handler.rb

```diff
@@ -208,7 +207,7 @@ def detach_context(request)
           end
 
           def add_response_attributes(span, response)
-            span.status = OpenTelemetry::Trace::Status.error unless GOOD_HTTP_STATUSES.include?(response.status.to_i)
+            span.status = OpenTelemetry::Trace::Status.error if response.server_error?
```

Comment:
I found out that here we have access to the final [Rack::Response](https://github.com/rack/rack/blob/8f1fe7d6b477d460b2365e2aadc2b348639f26f7/lib/rack/response.rb#L187) object so this existing `server_error?` method can be re-used (not the case in the middleware instrument version though)